### PR TITLE
Talentlms Tests

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
Add rspec tests for the talentlms facade model

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [X] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

**Comment by eanders** [comment]
@dtgreiner looks like this isn't working as expected on the CI.  Maybe it needs some adjustment to the VCR config?

**Comment by dtgreiner** [comment]
> @dtgreiner looks like this isn't working as expected on the CI. Maybe it needs some adjustment to the VCR config?

yeah, it shouldn't have even been tying to hit the api, or at lest wasn't in my local runs. I'll dig more into it and see what I can find.

File: spec/models/talentlms/facade_spec.rb [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+      login = Talentlms::Login.where(user: user).first
+      expect([login.login, login.lms_user_id]).to eq([DEFAULT_LMS_USERNAME, DEFAULT_LMS_USER_ID])
+      expect(user.talent_lms_email).to eq(DEFAULT_LMS_EMAIL)
+    end
+
+    it 'local data is updated when user data does not match api user data' do
+      stub_so_api_user_does_not_matches_local_user(config: config)
+
+      login = Talentlms::Login.where(user: user).first
+      expect([login.login, login.lms_user_id]).to eq([DEFAULT_LMS_USERNAME, DEFAULT_LMS_USER_ID])
```
Comment (eanders) [inline spec/models/talentlms/facade_spec.rb:91]: should this also:
```
expect(user.talent_lms_email).to_not eq(UPDATED_LMS_EMAIL)
```

**Review by eanders — COMMENTED** [review]
Thanks for the thorough testing @dtgreiner !  Looks great, approved once tests are passing.

File: spec/models/talentlms/facade_spec.rb (hunk 1 above)
Comment (dtgreiner) [reply spec/models/talentlms/facade_spec.rb:91]: This has been added

**Comment by dtgreiner** [comment]
@eanders All the tests here are passing, and it's ready for you to review when you have the chance. Thanks!

**Review by eanders — APPROVED** [review]
Thanks @dtgreiner !

**Pull request merged**